PurchasesCoreSwift/Networking/Backend.swift

```diff
@@ -0,0 +1,650 @@
+        }
+    }
+
+    @objc(getOfferingsForAppUserID:completion:)
+    public func getOfferings(appUserID: String, completion: @escaping OfferingsResponseHandler) {
+        guard let appUserID = try? escapedAppUserID(appUserID: appUserID) else {
+            completion(nil, ErrorUtils.missingAppUserIDError())
+            return
+        }
+
+        let path = "/subscribers/\(appUserID)/offerings"
+        if add(callback: completion, key: path) {
+            return
+        }
+
+        httpClient.performGETRequest(serially: false, path: path, headers: authHeaders) { statusCode, response, error in
+            if error == nil && statusCode < HTTPStatusCodes.redirect.rawValue {
+                for callback in self.getOfferingsCallbacksAndClearCache(forKey: path) {
+                    callback(response, nil)
+                }
+                return
+            }
+
+            let errorForCallbacks: Error
+            if let error = error {
+                errorForCallbacks = ErrorUtils.networkError(withUnderlyingError: error)
+            } else if statusCode > HTTPStatusCodes.redirect.rawValue {
+                let backendCode = self.maybeNumberFromError(code: response?["code"])
+                let backendMessage = response?["message"] as? String
+                errorForCallbacks = ErrorUtils.backendError(withBackendCode: backendCode,
+                                                            backendMessage: backendMessage)
+            } else {
+                errorForCallbacks = ErrorUtils.unexpectedBackendResponseError()
+            }
+
+            for callback in self.getOfferingsCallbacksAndClearCache(forKey: path) {
+                callback(nil, errorForCallbacks)
+            }
+        }
+    }
+
+    @objc(getIntroEligibilityForAppUserID:receiptData:productIdentifiers:completion:)
+    public func getIntroEligibility(appUserID: String,
+                                    receiptData: Data,
+                                    productIdentifiers: [String],
+                                    completion: @escaping IntroEligibilityResponseHandler) {
+        guard productIdentifiers.count > 0 else {
+            completion([:], nil)
+            return
+        }
+
+        if receiptData.count == 0 {
+            if SystemInfo.isSandbox {
+                Logger.appleWarning(Strings.receipt.no_sandbox_receipt_intro_eligibility)
+            }
+
+            var eligibilities: [String: IntroEligibility] = [:]
+
+            for productID in productIdentifiers {
+                eligibilities[productID] = IntroEligibility(eligibilityStatus: .unknown)
+            }
+
+            completion(eligibilities, nil)
+            return
+        }
+
+        // Closure we can use for both missing appUserID as well as server error where we have an unknown
+        // eligibility status.
+        let unknownEligibilityClosure: () -> [String: IntroEligibility] = {
```

Comment:
perhaps we can have a `handleResult` method like we do for some of the others?

Reply:
💯